pyomo/contrib/solver/solvers/gurobi_persistent.py

```diff
@@ -687,6 +627,8 @@ def _add_constraints(self, cons: List[ConstraintData]):
 
     def _add_sos_constraints(self, cons: List[SOSConstraintData]):
         for con in cons:
+            # NOTE: This seems like an unnecessary local variable because it's
+            # never used or called again. But when removed, it causes KeyErrors.
```

Comment:
Haha! The line `self._symbol_map.removeSymbol(con)` in `_remove_sos_constraints` has to be removed with it.